tests/unit/test_partition_builder.py

```diff
@@ -0,0 +1,491 @@
+            yaml_configs_list[ind]["partitions"].append(
+                {"target_file_name": target_file_name, "yaml_config": yaml_config}
+            )
+
+    return yaml_configs_list
+
+
+def _get_fake_json_data(data):
+    for row in data:
+        row["date_value"] = str(row["date_value"])
+        row["timestamp_value"] = str(row["timestamp_value"])
+        row["text_constant"] = row["text_constant"]
+        row["text_numeric"] = row["text_numeric"]
+        row["text_value"] = row["text_value"]
+        row["text_value_two"] = row["text_value_two"]
+
+    return json.dumps(data)
+
+
+def _create_table_file(table_path, data):
+    """Create JSON File"""
+    with open(table_path, "w") as f:
+        f.write(data)
+
+
+def test_import(module_under_test):
+    assert module_under_test is not None
+
+
+def test_class_object_creation(module_under_test):
+    """Create a PartitionBuilder object passing 2 tables and assert the following
+    1. Table/Configs count
+    2. config_dir value present and absent
+    3. primary_key value
+    4. partition_key value present and absent
+    """
+    mock_config_manager1 = ConfigManager(
+        GENERATE_TABLE_PARTITIONS_CONFIG,
+        MockIbisClient(),
+        MockIbisClient(),
+        verbose=False,
+    )
+    mock_config_manager2 = ConfigManager(
+        GENERATE_TABLE_PARTITIONS_CONFIG,
+        MockIbisClient(),
+        MockIbisClient(),
+        verbose=False,
+    )
+    config_managers = [mock_config_manager1, mock_config_manager2]
+
+    parser = cli_tools.configure_arg_parser()
+
+    # partition_key is present and different from primary_key
+    # config_dir is passed
+    args = parser.parse_args(CLI_ARGS)
+    builder = module_under_test.PartitionBuilder(config_managers, args)
+    assert builder.table_count == len(config_managers)
+    assert builder.partition_key == "bike_id"
+    assert builder.primary_key == "station_id"
+
+    # partition_key is absent, expected to default to primary_key
+    args = parser.parse_args(CLI_ARGS_ABSENT)
+    builder = module_under_test.PartitionBuilder(config_managers, args)
+    assert builder.table_count == len(config_managers)
+    assert builder.partition_key == builder.primary_key
+
+
+def test_get_partition_key_filters(module_under_test):
+    """Build partitions filters and assert:
+    1. Table count
+    2. Filters count
+    3. Partition filters
+    """
+    data = _generate_fake_data(rows=1001, second_range=0)
+
+    source_json_data = _get_fake_json_data(data)
+    target_json_data = _get_fake_json_data(data)
+
+    _create_table_file(SOURCE_TABLE_FILE_PATH, source_json_data)
+    _create_table_file(TARGET_TABLE_FILE_PATH, target_json_data)
+
+    config_manager = _generate_config_manager("my_table")
+    config_managers = [config_manager]
+
+    parser = cli_tools.configure_arg_parser()
+    mock_args = parser.parse_args(CLI_ARGS_JSON_SOURCE)
+
+    mock_partition_filters = _generate_fake_partition_filters(
```

Comment:
Removed generation of `partition_filters_list` and `yaml_configs_list` via function.

Added expected `PARTITION_FILTERS_LIST` and `YAML_CONFIGS_LIST` to a json file `tests/unit/test_inputs/test_partition_builder.json` since expected `YAML_CONFIGS_LIST` is too large to store in `tests/unit/test_partition_builder.py` and would make it less readable.